src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Statistics/TableBuilderControllerTests.cs

```diff
@@ -19,145 +18,97 @@
-                    {
-                        "location", new List<LocationAttributeViewModel>
-                        {
-                            new()
-                            {
-                                Label = "A label",
-                                Value = "A value",
-                                Level = GeographicLevel.Region,
-                                Options = new List<LocationAttributeViewModel>
-                                {
-                                    new()
-                                    {
-                                        Id = Guid.NewGuid(),
-                                        Label = "A label",
-                                        Value = "A value",
-                                        GeoJson = true
-                                    }
-                                }
-                            }
-                        }
-                    }
+                    TimePeriod = "2020_AY"
                 },
-                PublicationName = "Publication name",
-                SubjectName = "Subject name",
-                GeoJsonAvailable = true,
-                TimePeriodRange = new List<TimePeriodMetaViewModel>
+                new()
                 {
-                    new(1234, TimeIdentifier.April)
-                    {
-                        Label = "A label"
-                    }
+                    TimePeriod = "2021_AY"
                 }
-            }
+            },
         };
 
         private readonly ObservationQueryContext _query = new()
         {
-            SubjectId = Guid.NewGuid()
+            SubjectId = SubjectId,
         };
 
         [Fact]
         public async Task Query()
         {
-            var cancellationToken = new CancellationToken();
+            var tableBuilderService = new Mock<ITableBuilderService>(Strict);
+            tableBuilderService
+                .Setup(s => s.Query(
+                    ReleaseId,
+                    ItIs.DeepEqualTo(ObservationQueryContext),
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(_tableBuilderResults);
 
-            var (controller, mocks) = BuildControllerAndDependencies();
+            var client = SetupApp(tableBuilderService: tableBuilderService.Object)
+                .CreateClient();
 
-            mocks.tableBuilderService
-                .Setup(s => s.Query(_releaseId, _query, cancellationToken))
-                .ReturnsAsync(_tableBuilderResults);
+            var response = await client.PostAsync(
+                $"/api/data/tablebuilder/release/{ReleaseId}",
+                new JsonNetContent(ObservationQueryContext));
 
-            var result = await controller.Query(_releaseId, _query, cancellationToken);
-            VerifyAllMocks(mocks);
+            VerifyAllMocks(tableBuilderService);
 
-            result.AssertOkResult();
+            response.AssertOk(_tableBuilderResults);
         }
 
```

Comment:
We should write a test for the CSV variant of this endpoint, otherwise we're not really proving this has been fixed.

See the data API equivalent endpoint(s) for examples.